tools/contrib/neon/neon_runner.py

```diff
@@ -0,0 +1,793 @@
+#! /usr/bin/env python
+ 
+"""
+|------------------------------------------------------------------|
+|---------------------  Instructions  -----------------------------|
+|------------------------------------------------------------------|
+This is a wrapper script for running CTSM simulation for one or more
+neon sites. 
+
+This script is only for neon site and we will develop a more general
+code later.
+
+This script first creates and builds a generic base case. 
+Next, it will clone the base_case for different neon sites and run
+types to reduce the need to build ctsm everytime. 
+
+This script will do the following:
+    1) Create a generic base case for cloning. 
+    2) Make the case for the specific neon site(s).
+    3) Make changes to the case, for:
+        a. AD spinup
+	    b. post-AD spinup
+        c. transient
+    	#---------------
+    	d. SASU or Matrix spinup
+    4) Build and submit the case.
+ 
+-------------------------------------------------------------------
+Instructions for running on Cheyenne/Casper:
+ 
+load the following into your local environment
+    module load python
+    ncar_pylib
+ 
+To remove NPL from your environment on Cheyenne/Casper:
+    deactivate
+ 
+-------------------------------------------------------------------
+To see the available options:
+    ./run_neon.py --help
+-------------------------------------------------------------------
+"""
+#TODO (NS)
+#- [ ]
+#- [ ] Case dependency and the ability to check case status
+#- [ ] If Case dependency works we don't need finidat given explicilty for post-ad and transient.
+
+#- [ ] checkout_externals instead of using env varaiable
+#- [ ] wget the fields available and run for those available
+ 
+#- [ ] Matrix spin-up if (SASU) Eric merged it in 
+#- [ ] Make sure both AD and SASU are not on at the same time
+
+#- [ ] Make sure CIME and other dependencies is checked out.
+
+ 
+#Import libraries
+from __future__ import print_function
+ 
+import os
+import sys
+import time 
+import shutil
+import logging
+import requests
+import argparse
+import subprocess
+import pandas as pd
+ 
+from datetime import date
+from getpass import getuser
+ 
+#-- get the environment variable
+cesmroot = os.environ.get('CESM_ROOT')
+
+cesmroot = '/home/negins/ctsm_cime/'
+
+_LIBDIR = os.path.join(cesmroot,"cime","scripts","Tools")
+sys.path.append(_LIBDIR)
+_LIBDIR = os.path.join(cesmroot,"cime","scripts","lib")
+sys.path.append(_LIBDIR)
```

Comment:
I like the use of cime case objects here. Thanks for your work on this @negin513 !

Once this is integrated with the other ctsm python modules, you can make use of https://github.com/ESCOMP/CTSM/blob/master/python/ctsm/path_utils.py and/or https://github.com/ESCOMP/CTSM/blob/master/python/ctsm/add_cime_to_path.py to add the cime lib to your path in what I think is a robust way. (I'm open to suggestions if you can see a better way to do this.)